ocw/lib/ec2.py

```diff
@@ -173,7 +173,11 @@ def delete_vpc_subnets(self, vpc) -> None:
                     self.log_info(f'Deletion of {interface} skipped due to dry_run mode')
                 else:
                     self.log_info(f'Deleting {interface}')
-                    interface.delete()
+                    try:
+                        interface.delete()
+                    except ClientError as exc:
```

Comment:
we can not just mute ALL ClientError's  and pretend that nothing happened . Only if you know some certain error you may mute but not ALL errors !

Reply:
no , in case of exception we should interrupt code execution . this is what people normally do when they don't know what happened . They stop and trying to figure out what just happened . And only when they know EXACTLY what happened they keep doing thing which just hit them again

Reply:
```
try:
            self.log_info(f"Delete {resource_type.title()} '{resource_name}'")
            response = request.execute()
            self.log_dbg(f"Deletion response: {response}")
            self.log_info(f"{resource_type.title()} '{resource_name}' deleted")
        except HttpError as err:
            if GCE.get_error_reason(err) == 'resourceInUseByAnotherResource':
                self.log_dbg(f"{resource_type.title()} '{resource_name}' can not be deleted because in use")
            elif GCE.get_error_reason(err) == 'badRequest':
                # These are system generated routes when you create a network. These
                # will be deleted by the deletion of the network and do not block the
                # deletion of that network.
                # There are no properties on the Route struct that indicate a route is a
                # default one. Typically, the name will contain the word "default" or the
                # description will contain the word "Default" but a property like Kind
                # returns "compute#route" for all routes.
                # All this creating false alarms in log which we want to prevent.
                # Only way to prevent is mute error
                if resource_type.title() == "Route" and "The local route cannot be deleted" in str(err):
                    self.log_info("Skip deletion of local route")
                else:
                    self.log_err(f"{resource_type.title()} '{resource_name}' can not be deleted. Error : {err}")
            else:
                raise err
```

this is example of what I mean . this is how it should be handled . Just saying " ClientError something ? Who cares let's move on " is not acceptable . we can do some bad things this way ...

Reply:
```
2023-08-26 04:04:23,798 ocw.lib.ec2  ERROR    [sapha] delete_vpc_subnets: %s                                                                                                                │
│2023-08-26 04:04:23,798 ocw.lib.ec2  INFO     [sapha] Deleting ec2.Subnet(id='subnet-0d08c1fd959621eb8') 
2023-08-27 04:01:49,036 ocw.lib.ec2  ERROR    [ccoe] [vpc-036259e4bc0e3c996] ClientError on VPC deletion. Traceback (most recent call last):                                                │
│  File "/pcw/ocw/lib/ec2.py", line 152, in delete_vpc                                                                                                                                       │
│    self.delete_security_groups(vpc)                                                                                                                                                        │
│  File "/pcw/ocw/lib/ec2.py", line 217, in delete_security_groups                                                                                                                           │
│    sgroup.delete()                                                                                                                                                                         │
│  File "/usr/lib/python3.11/site-packages/boto3/resources/factory.py", line 580, in do_action                                                                                               │
│    response = action(self, *args, **kwargs)                                                                                                                                                │
│               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                │
│  File "/usr/lib/python3.11/site-packages/boto3/resources/action.py", line 88, in __call__                                                                                                  │
│    response = getattr(parent.meta.client, operation_name)(*args, **params)                                                                                                                 │
│               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                 │
│  File "/usr/lib/python3.11/site-packages/botocore/client.py", line 535, in _api_call                                                                                                       │
│    return self._make_api_call(operation_name, kwargs)                                                                                                                                      │
│           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                      │
│  File "/usr/lib/python3.11/site-packages/botocore/client.py", line 980, in _make_api_call                                                                                                  │
│    raise error_class(parsed_response, operation_name)                                                                                                                                      │
│botocore.exceptions.ClientError: An error occurred (DependencyViolation) when calling the DeleteSecurityGroup operation: resource sg-006564456452a5987 has a dependent object               │
│  File "/pcw/ocw/lib/ec2.py", line 217, in delete_security_groups                                                                                                                           │
│                                                                                                                                                                                            │
│    sgroup.delete()                                                                                                                                                                         │
│                                                                                                                                                                                            │
│  File "/usr/lib/python3.11/site-packages/boto3/resources/factory.py", line 580, in do_action                                                                                               │
│                                                                                                                                                                                            │
│    response = action(self, *args, **kwargs)                                                                                                                                                │
│                                                                                                                                                                                            │
│               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                │
│                                                                                                                                                                                            │
│  File "/usr/lib/python3.11/site-packages/boto3/resources/action.py", line 88, in __call__                                                                                                  │
│                                                                                                                                                                                            │
│    response = getattr(parent.meta.client, operation_name)(*args, **params)                                                                                                                 │
│                                                                                                                                                                                            │
│               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                 │
│                                                                                                                                                                                            │
│  File "/usr/lib/python3.11/site-packages/botocore/client.py", line 535, in _api_call                                                                                                       │
│                                                                                                                                                                                            │
│    return self._make_api_call(operation_name, kwargs)                                                                                                                                      │
│                                                                                                                                                                                            │
│           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                      │
│                                                                                                                                                                                            │
│  File "/usr/lib/python3.11/site-packages/botocore/client.py", line 980, in _make_api_call                                                                                                  │
│                                                                                                                                                                                            │
│    raise error_class(parsed_response, operation_name)                                                                                                                                      │
│                                                                                                                                                                                            │
│botocore.exceptions.ClientError: An error occurred (DependencyViolation) when calling the DeleteSecurityGroup operation: resource sg-006564456452a5987 has a dependent object
```

this is log from prod with PCW which has this patch applied . seems like this patch not helping :thinking: